CHANGELOG.md

```diff
@@ -86,6 +86,117 @@
   `Options::as<T>()`
   [\#2341](https://github.com/boutproject/BOUT-dev/pull/2341)
 
+## v4 known bugs
+
+- `FieldPerp` I/O has a couple of issues:
+  - the `global_yindex` attribute written in output files does not account for
+    boundary cells at the second (i.e.  upper) targets in double-null
+    configuration. This will be fixed in v5 using updated (but not
+    backward-compatible) indexing methods from `Mesh`.
+  - A `FieldPerp` may be read in on processors at different (but adjacent)
+    y-locations as it may be read in the guard cells. This is not the expected
+    behaviour (a given `FieldPerp`, if it represents a physical quantity rather
+    than just being a temporary variable for an x-z slice) should only be
+    defined and used at a single, global y-position. Should not be a problem
+    for `PhysicsModel` implementations that define a `FieldPerp` correctly, but
+    caused a problem for `test-io` and `test-io_hdf5`.
```

Comment:
I've forgotten what the 'not backward-compatible' thing was now, but did #2395 fix this bug for 4.4?

Reply:
Hmm, me either! I thought it was `getGlobalYIndex`/`getGlobalYIndexNoBoundaries`, but it looks like this did get merged into v4.3 at some point, so now I'm not sure.

Reply:
I'm confused. When I added the note in #2154, I claimed that we didn't have the updated `getGlobalYIndex()` method in the v4 branch, but it looks very much like we did, because it was added in #1770 which went into v4.3.0. I don't remember at all why I wrote that part now. Probably best to delete it (I guess from `master` and here?)
```suggestion
- A `FieldPerp` may be read in on processors at different (but adjacent)
  y-locations as it may be read in the guard cells. This is not the expected
  behaviour (a given `FieldPerp`, if it represents a physical quantity rather
  than just being a temporary variable for an x-z slice) should only be
  defined and used at a single, global y-position. Should not be a problem
  for `PhysicsModel` implementations that define a `FieldPerp` correctly, but
  caused a problem for `test-io` and `test-io_hdf5`.
```
I'm not sure if the second issue is still correct or not, probably worth checking that too...

Reply:
yeah, looks like the second one is fixed by the `FieldPerp::getGlobalIndex()` method from #2395, so we can delete that too.
```suggestion
```